lib/resourcemerge/apiext.go

```diff
@@ -12,18 +12,36 @@ import (
 // modified is set to true when existing had to be updated with required.
 func EnsureCustomResourceDefinitionV1(modified *bool, existing *apiextv1.CustomResourceDefinition, required apiextv1.CustomResourceDefinition) {
 	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
+	ensureCustomResourceDefinitionV1Defaults(&required)
+	ensureCustomResourceDefinitionV1CaBundle(&required, *existing)
+	if !equality.Semantic.DeepEqual(existing.Spec, required.Spec) {
+		*modified = true
+		existing.Spec = required.Spec
+	}
+}
 
-	// apply defaults to blue print
+func ensureCustomResourceDefinitionV1Defaults(required *apiextv1.CustomResourceDefinition) {
 	if len(required.Spec.Names.Singular) == 0 {
 		required.Spec.Names.Singular = strings.ToLower(required.Spec.Names.Kind)
 	}
 	if len(required.Spec.Names.ListKind) == 0 {
 		required.Spec.Names.ListKind = required.Spec.Names.Kind + "List"
 	}
+}
 
-	// we stomp everything
-	if !equality.Semantic.DeepEqual(existing.Spec, required.Spec) {
-		*modified = true
-		existing.Spec = required.Spec
+// ensureCustomResourceDefinitionV1CaBundle ensures that the field
+// .Spec.Conversion.Webhook.ClientConfig.CABundle is not managed by the CVO when
+// the service-ca controller is responsible for the field.
+func ensureCustomResourceDefinitionV1CaBundle(required *apiextv1.CustomResourceDefinition, existing apiextv1.CustomResourceDefinition) {
+	if val, ok := existing.ObjectMeta.Annotations[InjectCABundleAnnotation]; !ok || val != "true" {
+		return
+	}
+	if required.Spec.Conversion != nil && required.Spec.Conversion.Strategy == apiextv1.WebhookConverter {
```

Comment:
This is something new to me. So will dig more and get back.

Reply:
Looking at https://docs.openshift.com/container-platform/4.11/security/certificates/service-serving-certificate.html#add-service-certificate-crd_service-serving-certificate this PR looks legit to me. Many CRD manifests in the payload have that annotation so I'd expect this documented behavior to happen on them.

Reply:
Yeah, I agree.